sql-plugin/src/main/scala/com/nvidia/spark/rapids/conditionalExpressions.scala

```diff
@@ -170,24 +128,12 @@ case class GpuCaseWhen(
   }
 
   override def columnarEval(batch: ColumnarBatch): Any = {
-    val elseExpr = elseValue.getOrElse(GpuLiteral(null, branches.last._2.dataType))
-    try {
-      branches.foldRight[Any](elseExpr) { case ((predicateExpr, trueExpr), falseObj) =>
-        falseObj match {
-          case v: GpuColumnVector =>
-            try {
-              computeIfElse(batch, predicateExpr, trueExpr, v)
-            } finally {
-              v.close()
-            }
-          case e: GpuExpression => computeIfElse(batch, predicateExpr, trueExpr, e)
-        }
-      }
-    } finally {
-      elseExpr match {
-        case a: AutoCloseable => a.close()
-        case _ =>
-      }
+    // `elseRet` will be closed in `computeIfElse`.
+    val elseRet = elseValue
+      .map(_.columnarEval(batch))
+      .getOrElse(GpuScalar(null, branches.last._2.dataType))
+    branches.foldRight[Any](elseRet) { case ((predicateExpr, trueExpr), falseRet) =>
```

Comment:
If you get an AutoClose there, it should be closed there after using instead of closing it in another function.

Reply:
I think it is ok since this `computeIfElse` is not public. We know all the places it is used. Besides closing it in `computeIfElse` can keep the code quite simple.